### PR TITLE
Various installation quality of life improvements

### PR DESCRIPTION
**Pull request by emmet-m**
* Make `cogent.cabal` require `cabal<3.0`
* Add better error messages to `Setup.hs`
* Add some troubleshooting docs

### CODE REVIEW

**Comment by jashank** [comment]
As promised, fixes to `stack build`, and a partial clean-up of `man-gen.pl`.

File: cogent/Setup.hs [hunk 1]
```diff
@@ -102,15 +102,28 @@ copyGumHdrs verbosity pkg lbi copy = do
   let hdrsdest = datadir (L.absoluteInstallDirs pkg lbi copy) </> "include"
   putStrLn $ "Installing Gum headers in `" ++ hdrsdest ++ "'..."
   createDirectoryIfMissingVerbose verbosity True hdrsdest
-  installDirectoryContents verbosity "lib" hdrsdest
+  errorMsg <- installDirectoryContents verbosity "lib" hdrsdest
+  case errorMsg of 
+    Nothing -> return ()
+    Just x  -> do putStrLn x; putStrLn "Are you using cabal2.4?"
```
Comment (zilinc) [inline cogent/Setup.hs]: What's special in cabal 2.4? Also that thing is usually called `cabal-install-2.4`, to disambiguate from `Cabal-2.4`,

File: cogent/scripts/man-gen.pl [hunk 2]
```diff
@@ -1,87 +1,140 @@
...
 #
-# This software may be distributed and modified according to the terms of
-# the GNU General Public License version 2. Note that NO WARRANTY is provided.
-# See "LICENSE_GPLv2.txt" for details.
+# This software may be distributed and modified according to the
+# terms of the GNU General Public License version 2. Note that
+# NO WARRANTY is provided.  See "LICENSE_GPLv2.txt" for details.
 #
 # @TAG(NICTA_GPL)
 #
```
Comment (zilinc) [inline cogent/scripts/man-gen.pl]: We can change to the new header with the year of 2020 as this file has been almost rewritten:
```
--
-- Copyright 2020, Data61
-- Commonwealth Scientific and Industrial Research Organisation (CSIRO)
-- ABN 41 687 119 230.
--
-- This software may be distributed and modified according to the terms of
-- the GNU General Public License version 2. Note that NO WARRANTY is provided.
-- See "LICENSE_GPLv2.txt" for details.
--
-- @TAG(DATA61_GPL)
--
```

File: docs/install.rst [hunk 3]
```diff
@@ -217,3 +219,37 @@ A solution:
...
+  cabal: Could not resolve dependencies:
+  [__0] trying: cogent-2.9.0.0 (user goal)
+  [__1] unknown package: isa-parser (dependency of cogent)
+  [__1] fail (backjumping, conflict set: cogent, isa-parser)
+  After searching the rest of the dependency tree exhaustively, these were the
+  goals I've had most trouble fulfilling: cogent, isa-parser
+
+``isa-parser`` must be installed manually in this case. Change to the directory ``isa-parser`` at
+the root of the repository, and run ``cabal install``. Then, retry installing/building Cogent.
+
```
Comment (zilinc) [inline docs/install.rst]: If using `cabal v1-sandbox`, this should be done by `cabal v1-sandbox add-source ../isa-parser`, which I think is already included in the instruction and in the Makefile. Not sure about stack though.

**Comment by zilinc** [comment]
I'm happy with the changes. @jashank and @emmet-m, you can go ahead and merge when you feel appropriate.

**Pull request merged**